### PR TITLE
PR for SRVLOGIC-370: Create an initial PR that sets up the basic OSL "Supporting services" directory structure.

### PR DESCRIPTION
**Pull request by kaldesai**
cherrypick to `serverless-docs-1.34` per slack discussion.

**Notes for the peer and merge reviewers:**

As per the [slack discussion](https://redhat-internal.slack.com/archives/C063D46QWD9/p1721814386769389), I have created an initial PR that sets up the basic directory structure. Once this initial PR is merged, I will add the individual modules in smaller, incremental PRs. This way, I can avoid having one large PR and make the review process more manageable.

**Tracking JIRA:** https://issues.redhat.com/browse/SRVLOGIC-370

### CODE REVIEW

File: serverless-logic/serverless-logic-supporting-services/serverless-logic-data-index.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="serverless-logic-data-index"]
+= Data index
+include::_attributes/common-attributes.adoc[]
+:context: serverless-logic-data-index
+
+toc::[]
+
+The Data index service is a dedicated supporting service that stores the data related to the workflow instances and their associated jobs. This service provides a GraphQL endpoint allowing users to query and modify that data.
```
Comment (adellape) [inline serverless-logic/serverless-logic-supporting-services/serverless-logic-data-index.adoc:9]: And similar question here about whether "service" is the Kube API object (see comment on "Job service").

**Review by adellape — COMMENTED** [review]
Basic structure of this LGTM. My only questions would be around the marking up and/or capitalization of "Job service" and "Data index" but I didn't see any attached Eng tickets to the docs ticket to check how it looks elsewhere or in the code. Maybe something to consider as these new topics evolve.

File: serverless-logic/serverless-logic-supporting-services/serverless-logic-data-index.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaldesai) [reply serverless-logic/serverless-logic-supporting-services/serverless-logic-data-index.adoc:9]: Yes, I verified. 
Job Service and Data Index should be always be capitalized. and I will include the "service" in the title to maintain the consistency.

File: serverless-logic/serverless-logic-supporting-services/serverless-logic-job-service.adoc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="serverless-logic-job-service"]
+= Job service
+include::_attributes/common-attributes.adoc[]
+:context: serverless-logic-job-service
+
+toc::[]
+
+The Job service schedules and executes tasks in a cloud environment. Independent services implement these tasks, which can be initiated through any of the supported interaction modes, including HTTP calls or Knative Events delivery.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply serverless-logic/serverless-logic-supporting-services/serverless-logic-job-service.adoc:9]: It is not a a Service API object
In a nutshell, "Job services in Serverless Logic handle task execution." so it should generally remain lowercase.  So I will make the changes now. :) Thank you!

**Comment by bscott-rh** [comment]
Merge review LGTM.

**Pull request merged**